flax/nn/linear.py

```diff
@@ -219,12 +227,16 @@ def apply(self,
     """
 
     inputs = jnp.asarray(inputs, dtype)
+    if isinstance(kernel_size, int):
+      kernel_size = (kernel_size,)
 
     is_single_input = False
     if inputs.ndim == len(kernel_size) + 1:
       is_single_input = True
       inputs = jnp.expand_dims(inputs, axis=0)
-      
+
+    kernel_size = _make_correct_length(kernel_size, inputs.ndim)
```

Comment:
The broadcasting magic is not compatible with the code above recognizing whether the input has a batch dimension. `if inputs.ndim == len(kernel_size) + 1` needs the true kernel size to work correctly.

Reply:
Good point! But don't we need to know whether there is a batch size to infer the true kernel size if an int is passed? It's tricky to guess what the user means since there aren't separate Conv1D, Conv2D, ... classes. Maybe we should just issue an error when an array isn't passed in?

Reply:
So I think what we can do is interpret int as (int,) so basically int becomes a short-hand for 1D conv.
(I would prefer that for consistency with other int to tuple conversions anyway.)

Reply:
Done! Good idea.